compiler/src/parsing/parser.mly

```diff
@@ -308,7 +308,7 @@ typ:
 
 arg_typ:
   | LIDENT colon typ { TypeArgument.mk ~loc:(to_loc $loc) (Labeled (mkstr $loc($1) $1)) $3 }
-  | QUESTION LIDENT colon typ { TypeArgument.mk ~loc:(to_loc $loc) (Default (mkstr $loc($2) $2)) $4 }
+  | QUESTION LIDENT colon typ { TypeArgument.mk ~loc:(to_loc $loc) (Default ((mkstr $loc($2) $2), (Expression.constant Constant.void) (* dummy value *))) $4 }
```

Comment:
Just because it seemed to conform better for what I was trying to do. This `expression` is only read when printing the default value of a parameter, and this value would always be filled in properly when parsing a function declaration. If I were to use an option here instead there would still more or less be an `Option.get` used to get the value